### PR TITLE
Add cache layer docs

### PR DESCRIPTION
**Pull request by a-mccarthy**
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

fixes: https://github.com/goharbor/harbor/issues/17078

I've also made edits to the related Wiki page - https://github.com/goharbor/perf/wiki/Cache-layer. Please also review that page. Specifically, I've added more details to the architecture section to describe the diagram.

### CODE REVIEW

File: docs/install-config/configure-yml-file.md [hunk 1]
```diff
@@ -196,6 +196,21 @@ You can use certificates that are signed by a trusted third-party CA, or you can
     <td valign="top"><code>no_proxy</code></td>
     <td valign="top">Configure when not to use a proxy, for example, <code>127.0.0.1,localhost,core,registry</code>.</td>
   </tr>
+  <tr>
+    <td valign="top"><code>cache</code></td>
+    <td valign="top">&nbsp;</td>
+    <td valign="top">Configure cache layer for your Harbor instance. When enabled, Harbor will cache some Harbor resources (for example, artifacts, projects, or project metadata) using Redis, reducing the amount of time and resources used for repeated requests for the same Harbor resource. It's strongly recommended that you enable this feature on Harbor instances with high pull request rates to improve Harbor's overall performance. For more details on the cache layer implementation and performance improvements, see the the <a href="https://github.com/goharbor/perf/wiki/Cache-layer">Cache Layer wiki page</a>.</td>
```
Comment (chlins) [inline docs/install-config/configure-yml-file.md:202]: 1. `...with high concurrent pull...`
2. duplicate `the` after `see...`?

**Review by chlins — APPROVED** [review]
lgtm

File: docs/install-config/configure-yml-file.md (hunk 1 above)
Comment (a-mccarthy) [reply docs/install-config/configure-yml-file.md]: thanks @chlins!

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**